src/tensorflow.cc

```diff
@@ -698,8 +723,228 @@ class ModelState : public BackendModel {
+        graph_level, BackendConfig()->allow_gpu_memory_growth_,
+        BackendConfig()->per_process_gpu_memory_fraction_,
+        BackendConfig()->allow_soft_placement_,
+        BackendConfig()->memory_limit_mb_, tftrt_config_ptr,
+        auto_mixed_precision));
+    lmodel_handle.reset(model);
+
+    RETURN_IF_ERROR(
+        graphdef::ValidateTRITONTFModel(Name(), ModelConfig(), model));
+  } else {
+    TRITONTF_Model* model = nullptr;
+    RETURN_IF_TRITONTF_ERROR(TRITONTF_ModelCreateFromSavedModel(
+        &model, Name().c_str(), model_path.c_str(), device_id, has_graph_level,
+        graph_level, BackendConfig()->allow_gpu_memory_growth_,
+        BackendConfig()->per_process_gpu_memory_fraction_,
+        BackendConfig()->allow_soft_placement_,
+        BackendConfig()->memory_limit_mb_, tftrt_config_ptr,
+        auto_mixed_precision));
+    lmodel_handle.reset(model);
+
+    RETURN_IF_ERROR(savedmodel::ValidateTRITONTFModel(
+        Name(), ModelConfig(), MaxBatchSize(), model, &(lmodel.input_name_map_),
+        &(lmodel.output_name_map_)));
+  }
+
+  if (lmodel.input_device_id_ != ModelState::MODEL_DEVICE) {
+    std::vector<const char*> input_names, output_names;
+    std::vector<TRITONTF_DataType> input_types, output_types;
+    std::deque<std::string> io_names;
+
+    triton::common::TritonJson::Value config_inputs;
+    RETURN_IF_ERROR(ModelConfig().MemberAsArray("input", &config_inputs));
+    for (size_t i = 0; i < config_inputs.ArraySize(); i++) {
+      triton::common::TritonJson::Value io;
+      RETURN_IF_ERROR(config_inputs.IndexAsObject(i, &io));
+      io_names.emplace_back();
+      RETURN_IF_ERROR(io.MemberAsString("name", &io_names.back()));
+      std::string io_data_type;
+      RETURN_IF_ERROR(io.MemberAsString("data_type", &io_data_type));
+
+      input_names.push_back(io_names.back().c_str());
+      input_types.push_back(ConvertDataType(io_data_type));
+    }
+
+    triton::common::TritonJson::Value config_outputs;
+    RETURN_IF_ERROR(ModelConfig().MemberAsArray("output", &config_outputs));
+    for (size_t i = 0; i < config_outputs.ArraySize(); i++) {
+      triton::common::TritonJson::Value io;
+      RETURN_IF_ERROR(config_outputs.IndexAsObject(i, &io));
+      io_names.emplace_back();
+      RETURN_IF_ERROR(io.MemberAsString("name", &io_names.back()));
+      std::string io_data_type;
+      RETURN_IF_ERROR(io.MemberAsString("data_type", &io_data_type));
+
+      output_names.push_back(io_names.back().c_str());
+      output_types.push_back(ConvertDataType(io_data_type));
+    }
+    RETURN_IF_TRITONTF_ERROR(TRITONTF_ModelMakeCallable(
+        lmodel_handle.get(), input_names.data(), input_types.data(),
+        config_inputs.ArraySize(), output_names.data(), output_types.data(),
+        config_outputs.ArraySize()));
+  }
+
+  lmodel.tritontf_model_ = lmodel_handle.get();
+  model_handles_.emplace_back(std::move(lmodel_handle));
+  models_[device_id] = std::make_pair(1, lmodel);
```

Comment:
So, after every `max_session_share_count_` (and first session creation) the models_[device_id] is updated to 1. So, the counting can begin anew. The logic looks little separated from top. Will be better to include `Model lmodel` initialization(L743 to L939) into separate function to make this easier to read and understand.

Reply:
This is already the function for initializing the model. Other than that, there is only the top and the bottom where it updates the corresponding members in ModelState

Reply:
Rename the current CreateModel to GetModel and include the section I pointed into a new abstraction called CreateModel().